vault/logical_system_test.go

```diff
@@ -2420,6 +2420,62 @@ func TestSystemBackend_enableAudit(t *testing.T) {
 	}
 }
 
+func missingPayloadTest(path string, data map[string]interface{}) func(t *testing.T) {
+	return func(t *testing.T) {
+		_, b, _ := testCoreSystemBackend(t)
+		req := logical.TestRequest(t, logical.UpdateOperation, path)
+		req.Data = data
+		resp, err := b.HandleRequest(namespace.RootContext(nil), req)
+		if err == nil {
+			t.Fatalf("err: %v", err)
+		}
+		schema.ValidateResponse(
+			t,
+			schema.GetResponseSchema(t, b.(*SystemBackend).Route(req.Path), req.Operation),
+			resp,
+			true,
+		)
+	}
+}
+
+func TestSystemBackend_decodeToken(t *testing.T) {
+	encodedToken := "Bxg9JQQqOCNKBRICNwMIRzo2J3cWCBRi"
+	otp := "3JhHkONiyiaNYj14nnD9xZQS"
+	tokenExpected := "4RUmoevJ3lsLni9sTXcNnRE1"
+
+	t.Run("basic", func(t *testing.T) {
```

Comment:
I don't object to the test as written, but since there's no side effect from the new endpoint, the test could be written to share the backend/core returned by testCoreSystemBackend across all four subtests, rather than creating a fresh one for each.  This would be a little more performant and conventional, but I don't mind if you choose not to do this, it's more of a suggestion than a request.

Reply:
I liked to try out returning a function by a function. I understand the performance suggestion and refactored the test.